ganga/GangaCore/Lib/Localhost/LocalHostExec.py.template

```diff
@@ -85,6 +88,27 @@ errorfile=open('stderr','w')
 sys.stdout=open('./__syslog__','w')
 sys.stderr=sys.stdout
 
+from Virtualization import checkDocker, checkUDocker, checkSingularity, installUdocker
+
+if (virtualization == "Docker"):
+    appscriptpath[0] = "/work_dir/"+appscriptpath[0]
+    if (checkDocker()):
+        print("Using Docker")
+        appscriptpath = ['docker', 'run', '-v', workdir+":"+"/work_dir", virtualization_image] + appscriptpath        
+    else:
+        print("Docker not availabe or user doesn't have the permission to run docker demon")
+        if not (checkUDocker()):
+            installUdocker()
+        print("Using UDocker")
+        os.environ["PROOT_NO_SECCOMP"] = "1"
+        appscriptpath = ['/home/inzamam/udocker', '--quiet', 'run', '--nosysdirs', '--volume', workdir+":"+"/work_dir", virtualization_image] + appscriptpath
```

Comment:
This needs to be pointed at wherever udocker is installed rather than hard coded to your home dir which wont exist if I (or anyone else) run it.

Reply:
oh sorry it's a mistake :)